substrate/frame/salary/src/lib.rs

```diff
@@ -447,3 +449,21 @@ pub mod pallet {
 		}
 	}
 }
+
+impl<T: Config<I>, I: 'static>
+	RankedMembersSwapHandler<T::AccountId, <T::Members as RankedMembers>::Rank> for Pallet<T, I>
+{
+	fn swapped(
+		who: &T::AccountId,
+		new_who: &T::AccountId,
+		_rank: <T::Members as RankedMembers>::Rank,
+	) {
+		let Some(claimant) = Claimant::<T, I>::take(&who) else {
+			frame_support::defensive!("Claimant should exist when swapping");
+			return;
+		};
+
+		Claimant::<T, I>::insert(&new_who, &claimant);
```

Comment:
I saw that caller of `swapped` in `pallet-ranked-collective` checks that `new_who` isn't already stored so it doesn't get overwritten, but should we not also do a check here? It's hard to ensure the state is consistent between two or more pallets dealing with the swaps.